.github/workflows/chromatic.yaml

```diff
@@ -30,13 +26,36 @@ jobs:
       - run: yarn install --frozen-lockfile
         env:
           NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - uses: dorny/paths-filter@v2
+        id: changes
+        with:
+          filters: |
+            packages:
+              - 'packages/**'
+            draft:
+              - 'draft-packages/**'
 
       # 👇 Adds Chromatic as a step in the workflow
+
+      # Run when push event and review state is approved
+      - name: Publish to Chromatic - Merge Master into branch
+        if: github.event.review.state == 'approved' && (github.ref == 'refs/heads/master' || contains(github.event.head_commit.message, "Merge branch 'master' into")
+        uses: chromaui/action@v1
+        # Chromatic GitHub Action options
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          # 👇 Chromatic projectToken, refer to the manage page to obtain it.
+          projectToken: ${{ secrets.CHROMATIC_PROJECT_TOKEN }}
+          autoAcceptChanges: master # 👈 Auto accept master builds (why https://www.chromatic.com/docs/github-actions#github-squashrebase-merge-and-the-main-branch)
+
+      # Run when 'package' or 'draft-packages' files where changes
       - name: Publish to Chromatic
+        if: (steps.changes.outputs.packages == 'true' || steps.changes.outputs.draft == 'true') || (github.event.action == 'labeled' && github.event.label.name == 'run chromatic')
```

Comment:
If files changes are within package or draft package - Run
If a label of run chromatic has been added - run